packages/autorest.python/autorest/codegen/models/code_model.py

```diff
@@ -76,6 +76,19 @@ def __init__(
         self.named_unions: List[CombinedType] = [
             t for t in self.types_map.values() if isinstance(t, CombinedType) and t.name
         ]
+        self._has_etag: Optional[bool] = None
+
+    @property
+    def has_etag(self) -> bool:
+        if self._has_etag is None:
```

Comment:
this can either be done in preprocesser or here. I think it'd be cleaner if you just did a double list iteration,
```python
return any(
op.has_etag
for client in self.clients
for og in client.operation_groups
for op in og.operations
)
```

Reply:
We shouldn't need to initialize with a private `self._has_etag` as well. We shouldn't do that in code model or in operations

Reply:
Agree this could be moved to preprocessor.